src/pykeen/models/unimodal/complex.py

```diff
@@ -128,6 +128,13 @@ def interaction_function(
         :return: shape: (...)
             The scores.
         """
+        if t.ndim != h.ndim:
```

Comment:
this is pretty general logic, right? @mberr was also thinking about this sort of abstraction in #107

Reply:
I changed to re-using `get_in_canonical_shape` for this :slightly_smiling_face: